app/controllers/application_controller.rb

```diff
@@ -0,0 +1,10 @@
+class ApplicationController < ActionController::Base
+  before_action :set_locale
+  def set_locale
+    I18n.locale = params[:locale] || I18n.default_locale
+  end
+
+  def default_url_options
+    {locale: I18n.locale}
+  end
```

Comment:
private cho 2 method này nhé
```suggestion

  private
  
  def set_locale
    I18n.locale = params[:locale] || I18n.default_locale
  end

  def default_url_options
    {locale: I18n.locale}
  end
```